h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -382,11 +402,56 @@ public static Vec categoricalDomainsToNumeric(final Vec src) {
       }.doAll(Vec.T_NUM, src).outputFrame().anyVec();
   }
 
+  public static class CollectDoubleDomain extends MRTask<CollectDoubleDomain> {
```

Comment:
In the extreme case of a constant chunk we need to iterate over all values just to use one number in the end. I was just wondering is something like that would help also elsewhere. In a similar way we can iterate over non-zero/non-na elements.